marshmallow_dataclass/__init__.py

```diff
@@ -639,12 +658,27 @@ def _field_for_generic_type(
     """
     If the type is a generic interface, resolve the arguments and construct the appropriate Field.
     """
-    origin = typing_inspect.get_origin(typ)
-    arguments = typing_inspect.get_args(typ, True)
+    origin = typing_extensions.get_origin(typ)
+    arguments = typing_extensions.get_args(typ)
     if origin:
         # Override base_schema.TYPE_MAPPING to change the class used for generic types below
         type_mapping = base_schema.TYPE_MAPPING if base_schema else {}
 
+        if origin is Annotated:
+            marshmallow_annotations = [
+                arg
+                for arg in arguments[1:]
+                if (inspect.isclass(arg) and issubclass(arg, marshmallow.fields.Field))
+                or isinstance(arg, marshmallow.fields.Field)
+            ]
+            if marshmallow_annotations:
+                field = marshmallow_annotations[-1]
```

Comment:
We could publish a warning? The issue is that since annotations are just that, theoretically there could be other libraries that would use the multiple fields.  So I wouldn't want to raise an exception.

Reply:
> We could publish a warning? The issue is that since annotations are just that, theoretically there could be other libraries that would use the multiple fields. So I wouldn't want to raise an exception.

To be clear, the case I'm concerned with is when a type is annotated with multiple instances and/or subclasses of `marshmallow.fields.Field`.  Extra non-`Field` annotations do not cause any trouble — they will just be ignored here.
A `Field` instance's purpose is limited to specifying how to serialize/deserialize the value.  Other libraries, I think, will have as much trouble interpreting the meaning of multiple `Field` annotations as we do here.
E.g. Marshmallow itself does not provide a way to attach multiple `Field`s to a field — and if it did, what would it mean?

Since there seems to be no obvious way to interpret the semantics of multiple `Field` annotations other than to ignore all but one, it seems much safer and less potentially confusing to make it an error rather than a warning. 

But I guess I could live with it being a warning.

Reply:
I can't think of an actual reason why you'd specify multiple Fields. But it's not breaking. So I see it as we're making a decision of how we're handling it and telling the user about that decision.

But frankly, I looked at FastAPI and this is how they handle the same usecase. And I just copied their logic :)